### ISSUE TITLE
duplicate-code in Inheritance and implementation classes

### ISSUE THREAD

**Issue by whg517** [PyCQA/pylint#4350]
### question

When I inherited and implemented my superclass, PyLint detected duplication of part of the code that I defined for method parameters, and I wondered if PyLint could disable this.

What I'm saying is not to completely disallow duplicate-code, but to prevent PyLint from throwing warnings triggered by the above situation.

### Specific description:

When I have a base class that contains some public methods and some abstract methods, subclasses need to implement the abstract methods. In some cases, you may also override methods already implemented by the parent class.

So my subclass will integrate the base class and then implement the methods of the parent class. In general, there is no problem.

But when I use type annotations, I annotate the types for those parameters, and I annotate the return value of the method. The method definition is too long, so I have to use it

The indentation. The part of the method definition becomes multiline. This part of the code also needs to be reused for subclass integration. (When a subclass implements or overrides a parent's method, the method's parameters can be looser than the parent's, and the return value can be stricter.)

This causes PyLint to detect that multiple lines of code in the method definition section are duplicated. This part of the code should not actually throw a warning.

Of course I also tried to put this code on one line, and running PyLint doesn't throw a warning about duplicate code, but it does throw a warning that the trip is too long.

So I wanted to know how to solve this part of the problem.

### Specific cases:

**base.py**

```python
"""
Base
"""
import asyncio
from asyncio import AbstractEventLoop
from typing import Optional

class BaseExecutor:
    """
    Base executor.
    """
    def __init__(
        self,
        name: str,
        pid: str,
        loop: Optional[AbstractEventLoop] = None,
    ):
        self.name = name
        self.pid = pid
        self._loop = loop

    @property
    def loop(self) -> AbstractEventLoop:
        """Current event loop"""
        if self._loop:
            return self._loop
        return asyncio.get_running_loop()

    def run(self) -> None:
        """Run executor."""
        raise NotImplementedError

```

**executors.py**

```python
"""
Executors.
"""
from asyncio import AbstractEventLoop
from typing import Optional

from my_project.base import BaseExecutor


class DockerExecutor(BaseExecutor):
    """
    Docker Executor
    """
    def __init__(
        self,
        name: str,
        pid: str,
        loop: Optional[AbstractEventLoop] = None,
    ):
        super().__init__(name, pid, loop)

        self._client = 'Fake docker client'

    def run(self) -> None:
        """Run executor."""
        print(f'{id(self.loop)} Run executor')

```

**Pylint output:**

```
(my_project) ➜  my_project pylint src tests
************* Module tests.tests
tests/tests.py:1:0: R0801: Similar lines in 2 files
==src.my_project.base:11
==src.my_project.executors:13
    def __init__(
        self,
        name: str,
        pid: str,
        loop: Optional[AbstractEventLoop] = None,
    ): (duplicate-code)

------------------------------------------------------------------
Your code has been rated at 9.63/10 (previous run: 9.63/10, +0.00)
```

I'm also a little confused about the output. Why does tests/tests.py show that the two files are similar? I don't have any code under my tests.

**Comment by Pierre-Sassoulas** [comment]
You can use:

```ini
[SIMILARITIES]
# min-similarity= (more than the default 4 ?)
ignore-imports=yes
```

**Comment by Pierre-Sassoulas** [comment]
Feel free to reopen if that does not answer the question.

**Comment by whg517** [comment]
Hi @Pierre-Sassoulas 

Thank you for your reply.

This configuration is literally set to ignore imports, and it should not work in the case I mentioned above. Because the warning thrown in the above case is not related to the import code. As I suspected, I added the parameter `ignore-omports=yes` and the warning was not ignored.

[Here is my sample repository](https://github.com/whg517/demo) where I put the above code and automatically run PyLint through GitHub Action. In the pipeline, I first looked at the configuration of PyLint, and [I did enable it](https://github.com/whg517/demo/runs/2340869261?check_suite_focus=true#step:4:31), but the warning [still appears](https://github.com/whg517/demo/runs/2340869261?check_suite_focus=true#step:4:42).

Here is pylint config:

https://github.com/whg517/demo/blob/45b499dfa66235d5579802be95a38cfcb2dfb767/setup.cfg#L46

**Comment by whg517** [comment]
Hi @Pierre-Sassoulas . Could you please open it again? I have no authority to open the ISSUE you closed. Thanks a lot.

**Comment by Pierre-Sassoulas** [comment]
Sorry, I confused import with signature so in fact this is a dupe of #3619. You can disable similarities for this file as long as `ignore-signatures` this is not an available option in pylint.

**Comment by whg517** [comment]
I see, thank you, and hope the community can improve this feature as soon as possible.
